www/js/lib/zimfile.js

```diff
@@ -110,6 +110,42 @@ define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util,
         }
     };
 
+    /**
+     * Reads the whole MIME Type list and returns it as a populated Map
+     * 
+     * @return {Promise} A promise for the MIME Type list as a Map
+     */
+    ZIMFile.prototype.mimeTypeMap = function() {
+        var typeMap = this.mimeTypes;
+        // If we have already populated mimeTypes, we can just return it
+        if (typeMap.size) {
+            return Q.resolve().then(function() {
+                return typeMap;
+            });
+        } else {
+            // Otherwise, we have to populate it
+            return this._readSlice(this.mimeListPos, 256).then(function(data) {
```

Comment:
The function is set to read 256 bytes of data starting from `mimeListPos`. I have not found any MIME Type list that is longer than 130 bytes yet. Should we read 512 bytes, or even 1024 bytes, to be safe? Can we envisage any circumstance in which we have 1K of MIME Types defined?